app/src/main/java/com/github/se/polyfit/ui/screen/OverviewScreen.kt

```diff
@@ -316,6 +325,55 @@ fun OverviewScreen(
                   }
                 }
           }
+          item {
+            OutlinedCard(
+                modifier =
+                    Modifier.align(Alignment.Center)
+                        .padding(top = 10.dp)
+                        .size(width = 350.dp, height = 300.dp)
+                        .testTag("Graph Card")
+                        .clickable { navController.navigate(Route.Graph) },
+                border =
+                    BorderStroke(
+                        2.dp,
+                        brush =
+                            Brush.linearGradient(
+                                listOf(
+                                    MaterialTheme.colorScheme.inversePrimary,
+                                    MaterialTheme.colorScheme.primary))),
+                colors = CardDefaults.cardColors(Color.Transparent)) {
+                  Column(modifier = Modifier.fillMaxSize().testTag("Graph Card Column")) {
+                    Text(
+                        text = "Calories Graph",
+                        style = MaterialTheme.typography.headlineLarge,
+                        fontWeight = FontWeight.Bold,
+                        color = MaterialTheme.colorScheme.secondary,
+                        modifier =
+                            Modifier.padding(start = 10.dp, top = 10.dp)
+                                .weight(1f)
+                                .testTag("Graph Card Title"))
+                    if (!isTestEnvironment) {
+                      Box(
+                          modifier =
+                              Modifier.fillMaxSize(0.85f).align(Alignment.CenterHorizontally)) {
+                            LineChart(
+                                modifier =
+                                    Modifier.fillMaxWidth()
+                                        .testTag("Overview Line Chart")
+                                        .fillMaxSize()
+                                        .testTag("LineChart")
+                                        .clickable { navController.navigate(Route.Graph) }
+                                        .align(Alignment.Center),
+                                lineChartData =
+                                    lineChartData(
+                                        hiltViewModel<GraphViewModel>().DataPoints(),
+                                        hiltViewModel<GraphViewModel>().DateList(),
+                                        DisplayScreen.OVERVIEW))
+                          }
+                    }
+                  }
+                }
+          }
```

Comment:
Function too big so not readable. Like max 40 lines.

Reply:
this will be modified upon introduction OverviewScreenViewModel

Reply:
i agree with antoine - though not urgent, please make a ticket in the backlog to clean this debt later